plugins/disk/quota2percent_

```diff
@@ -1,143 +1,205 @@
+###################################################
+
+# Load munin's shell libary
+  . "$MUNIN_LIBDIR/plugins/plugin.sh"
+
+# if any fetch from munin-node file
+     Warning=${warning:-90}
+     Critical=${critical:-95}
+     Language=${language:-en}
+
+# Ensure that the 'root' path is valid
+  PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+
+# Checking if repquota installed and on the path
+  repquota -V &> /dev/null
+  if ! repquota -V &> /dev/null ; then
+     echo "The script 'repquota' is not installed or on the path"
+     # Send the exit code FATAL ERROR happens
+     exit 127
+  fi
 
-# Das Plugin zeigt die Festplattenbelegung in Prozent des Quota-Hard-Limits.
-# Munin plugin to show the disk usage in percent of the quota hard limit.
-# The base frame is copied from the proftp plugin
-
-##########################################################################################
-# Folgende Eintraege in der Datei  /etc/munin/plugin-conf.d/munin-node nicht vergessen ! # 
-# Don't forget to add following lines to the file /etc/munin/plugin-conf.d/munin-node    #
-# [quota2percent_*]                                                                      #
-# user root                                                                              #
-#                                                                                        #
-# Weiterhin sind folgende Environment-Variablen erlaubt:                                 #
-# Following environment variables are allowed:                                           #
-# env.warning   [value]                                                                  #
-# env.critical  [value]                                                                  #
-#                                                                 V17.0124  Jo Hartmann  #
-##########################################################################################
-
-# Spachauswahl 
-# Language selection
-  Language="de"
-
-
-# Wildcard-Text erkennen
 # get tehe wild card text
-  Id=$0
-  Id=${Id##*_}
+  Id=${0##*_}
+
+###################################################
+# Data reading sections                           #
+###################################################
 
-# Einlesen der Quoten für das entsprechende Laufwerk mit repquota 
 # Reading the quotes for the selected device, using repquota 
-   readarray Quotas < <( repquota /dev/$Id | grep " -- " )
-   readarray Totals < <(  df /dev/$Id )
+  readarray Quotas < <( repquota "/dev/$Id" | grep " -- " )
+  readarray Totals < <( df "/dev/$Id" )
 
-# Anzahl der Nutzer ermitteln 
 # Get the count of Users 
    Users=${#Quotas[@]}
 
 
 ###################################################
-# Beginn der Standard-Konfigurationsbereiches     #
-# Standard Config Section Begin                   #
+# Munin Configuration Section                     #
 ###################################################
 
   if [ "$1" = "autoconf" ]; then
        echo yes
```

Comment:
Your current approach of running the same check for all non-`autoconf` calls looks good to me.